presto-base-jdbc/src/test/java/io/prestosql/plugin/jdbc/TestCachingJdbcClient.java

```diff
@@ -105,10 +127,84 @@ public void testTableHandleCached()
         assertThat(cachingJdbcClient.getTableHandle(SESSION, phantomTable)).isEqualTo(cachedTable);
     }
 
+    @Test
+    public void shouldInvalidateTableCachesOnTableDrop()
+    {
+        SchemaTableName phantomTable = new SchemaTableName(schema, randomTableName());
```

Comment:
Can you please have two sessions (two users) cache table for both users and then when table is dropped in Presto both caches got invalidated. Also you can have third user to verify that other table do not get invalidated (optional).

Same goes to other tests like that.